### PR TITLE
fix: tour arrow color

### LINKED ISSUE #41760: Tour 漫游式引导的 type 为 primary 时，箭头的颜色仍为白色
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/ji-ben-antd-5-4-2-forked-ynr39y?file=/demo.tsx:433-456)

### Steps to reproduce

点击 Begin Tour

### What is expected?

该 step 或整个 Tour 的 type 为 primary 时，箭头应为蓝色。

### What is actually happening?

箭头颜色为白色。
image

| Environment | Info |
| --- | --- |
| antd | 5.4.2 |
| React | ^18.0.0 |
| System | macOS |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by iola1999**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #41760

### 💡 Background and solution

image

@heiyu4585 在 [PR#41761](https://github.com/ant-design/ant-design/pull/41761#discussion_r1165652834) 中提供的简洁解法，通过 selector 修复了。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix the problem that when the type of Tour is primary, the color of the arrow is still white      |
| 🇨🇳 Chinese |    修复 Tour 的 type 为 primary 时，箭头的颜色仍为白色的问题      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 22e9acf</samp>

Add `color` prop to `Tour` component and fix arrow color bug. This allows users to customize the color of the tour panels and arrows, and ensures that the arrow color matches the panel color when using a custom primary color.

### <samp>🤖 Generated by Copilot at 22e9acf</samp>

* Add a `color` prop to the `Tour` component to allow users to customize the color of the panels and arrows (F0L

### CODE REVIEW

File: components/tour/style/index.ts [hunk 1]
```diff
@@ -157,9 +157,10 @@ const genBaseStyle: GenerateStyle<TourToken> = (token) => {
 
         // =============================  primary type  ===========================
         // `$` for panel, `&$` for pure panel
-        [`${componentCls}-primary, &${componentCls}-primary`]: {
-          '--antd-arrow-background-color': colorPrimary,
-        },
+        [`${componentCls}-primary, &${componentCls}-primary, ${componentCls}-arrow:has(+ ${componentCls}-primary)`]:
```
Comment (afc163) [inline components/tour/style/index.ts:160]: primary 的状态范围还是应该放到 rootClassName 上，而不是用 selector 来 hack。

Comment (heiyu4585) [reply components/tour/style/index.ts:160]: 这个当初设计的时候,是有这个问题,感觉这块还是梳理一下,

Comment (MadCcc) [reply components/tour/style/index.ts:160]: `:has` 兼容性很差，chrome 要 105 版本以上才能使用

Comment (MadCcc) [reply components/tour/style/index.ts:160]: 直接选择自元素 arrow 就可以了吧

Comment (heiyu4585) [reply components/tour/style/index.ts:160]: 要根据后面兄弟元素的状态,去判断arrow的颜色, 直接选择是不是选不到

**Pull request closed without merging**